### PR TITLE
Fix #6: DONOTMERGE: Add react-redux typescript definition

### LINKED ISSUE #6: No d.ts file.
There isn't a typings file currently for `preact-redux`, which is a shame.

However it's only a micro layer around `react-redux`, which already has a typing, so it should be fairly easy to fix.

### PR DESCRIPTION
**Pull request by mseddon**
This is a pretty cheesy solution.  But hey!

Seems to have trouble with stateless functional components, need to parity check the support against react before it's ready, but at least it gives everyone a chance to have a look at what I'm doing.

### CODE REVIEW

**Comment by screendriver** [comment]
Would be great if this could be merged @developit :+1:

**Comment by developit** [comment]
That's the only reason I held off, otherwise this looks good to me (though I know little about TS)

**Comment by tobico** [comment]
I can confirm that this PR does not work with functional components. I played around a bit with pulling the latest definitions for react-redux, but they're quite complex and I couldn't understand them well enough to adapt them to preact.

Might be worth asking the authors of the react-redux bindings, @tkqubo @seansfkelley - would either of you be able to help get this working? Thanks!

**Comment by mseddon** [comment]
@screendriver yeah, stateless functions was the big one, various attempts to fix it triggered bazillions of errors.  I think I'd need a bit more Typescript-fu (and free time!) to get that working.  Any help appreciated!

**Comment by seansfkelley** [comment]
I am not familiar with preact or preact-redux, but compared to the [current react-redux typings](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react-redux/index.d.ts) there are more differences than I would expect. What kinds of errors do you get when you introduce stateless component types? Also, note that [Typescript cannot properly express the types of connect when used as a decorator](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/9951), so watch out for that.

**Comment by screendriver** [comment]
Any updates on this?

**Comment by rand0me** [comment]
Check out [#20](https://github.com/developit/preact-redux/pull/20), I'm using this typings and doesn't have any problems with functiopnal components

File: src/preact-redux.d.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+/// <reference types="preact" />
+/// <reference types="redux" />
+
+declare module "preact-redux" {
+  import { Component } from 'preact';
+  import { Store, Dispatch, ActionCreator } from 'redux';
+
+  export abstract class ElementClass extends Component<any, any> { }
+  export interface ClassDecorator {
+    <T extends (typeof ElementClass)>(component: T): T
```
Comment (geon) [inline src/preact-redux.d.ts]: Please use `typeof ElementClass | FunctionalComponent` instead. Preact supports fully functional components as well.

**Pull request closed without merging**